src/controllers/cost-analysis/cost/analyze-costs.ts

```diff
@@ -0,0 +1,190 @@
+import { cloneDeep } from 'lodash';
+import grpcClient from '@lib/grpc-client';
+
+const GRANULARITY_FORMAT = {
+    DAILY: '%Y-%m-%d',
+    MONTHLY: '%Y-%m',
+    YEARLY: '%Y',
+    ACCUMULATED: null
+};
+
+const getDefaultQuery: any = () => {
+    return {
+        query: {
+            aggregate: [
+                {
+                    group: {
+                        keys: [],
+                        fields: [
+                            {
+                                key: 'usd_cost',
+                                name: 'usd_cost',
+                                operator: 'sum'
+                            }
+                        ]
+                    }
+                }],
+            filter: []
+        }
+    };
+};
+
+const makeRequest = (params) => {
+    const requestParams = getDefaultQuery();
+
+    if (!params.start) {
+        throw new Error('Required Parameter. (key = start)');
+    }
+
+    if (!params.end) {
+        throw new Error('Required Parameter. (key = end)');
+    }
+
+    if (params.group_by) {
+        if (Array.isArray(params.group_by)) {
+            for (const groupKey of params.group_by) {
+                requestParams.query.aggregate[0].group.keys.push({
+                    key: groupKey,
+                    name: groupKey
+                });
+            }
+        }
+    }
+
+    if (Object.keys(GRANULARITY_FORMAT).indexOf(params.granularity) < 0) {
```

Comment:
```suggestion
    if (!Object.keys(GRANULARITY_FORMAT).includes(params.granularity)) {
```

이렇게 하면 가독성이 더 좋아질 것 같습니다!

Reply:
옙 다음 패치에 처리 완료했습니다.